### DISCUSSION TITLE
Deployment error with MySQL

### DISCUSSION THREAD

**Discussion by cristianoprogramador** [vercel/community#2466] [Help]
### Summary

I have an error when deploying the project because the error keeps appearing:

<html>
<body>

./node_modules/mysql2/lib/connection.js
--
21:43:56.500 | Module not found: Can't resolve 'net'
21:43:56.500 |  
21:43:56.500 | https://nextjs.org/docs/messages/module-not-found
21:43:56.500 |  
21:43:56.500 | Import trace for requested module:
21:43:56.500 | ./node_modules/mysql2/index.js
21:43:56.500 | ./node_modules/mysql2/promise.js
21:43:56.501 | ./src/pages/db.ts
21:43:56.501 |  
21:43:56.501 | ./node_modules/mysql2/lib/connection.js
21:43:56.501 | Module not found: Can't resolve 'tls'
21:43:56.501 |  
21:43:56.501 | https://nextjs.org/docs/messages/module-not-found
21:43:56.501 |  
21:43:56.501 | Import trace for requested module:
21:43:56.501 | ./node_modules/mysql2/index.js
21:43:56.501 | ./node_modules/mysql2/promise.js
21:43:56.501 | ./src/pages/db.ts
21:43:56.502 |  
21:43:56.502 | ./node_modules/mysql2/lib/server.js
21:43:56.502 | Module not found: Can't resolve 'net'

</body>
</html>

I don't know what else I need to configure, the project will work perfectly locally accessing my mysql database. Should I change from mysql12 to mysql?

**Comment by amyegan** [comment]
Hi @cristianoprogramador! If you share an example repo or steps we can use to reproduce the issue, someone here might be able to recognize what's causing the problem.

**Reply by cristianoprogramador** [reply]
oh ok, here is the repo for my project:
https://github.com/cristianoprogramador/mylife_dashboard
![image](https://github.com/vercel/community/assets/102186472/2b3110b1-7422-419b-91ca-0ddd9074af67)

**Reply by amyegan** [reply]
Thanks for sharing the repo! If I try to build it locally with `next build` or `npm run build`, I get the same errors. Does it build successfully in your local development environment?

**Reply by cristianoprogramador** [reply]
No it doesn't, I try to take it out the mysql2 it works the build, so I don't know if I'm putting the files in the right folder or if it's going to make a difference

**Reply by cristianoprogramador** [reply]
But it is working if i start the project locally with npm run dev, it works perfectly
the only issue is when i try "npm run build", this is driving me crazy lol

**Reply by amyegan** [reply]
It seems like there might be a configuration issue. If you haven't already reached out to the folks who maintain the `mysql2` package, you should try asking in https://github.com/sidorares/node-mysql2/discussions

You can find more info about using databases with Vercel in this guide: https://vercel.com/guides/using-databases-with-vercel#providers
